ihatemoney/currency_convertor.py

```diff
@@ -14,205 +21,93 @@ def __call__(cls, *args, **kwargs):
         return cls._instances[cls]
 
 
-class CurrencyConverter(object, metaclass=Singleton):
-    # Get exchange rates
-    no_currency = "XXX"
+class ExchangeRateGetter(ABC):
+
+    def get_rates(self) -> Optional[ExchangeRates]:
+        """Method to retrieve a list of currency conversion rates.
+
+        Returns:
+            currencies: dict - key is a three-letter currency, value is float of conversion to base currency
+        """
+        try:
+            return self._get_rates()
+        except Exception:
+            warnings.warn(
+                f"Exchange rate getter failed - {traceback.format_exc(limit=0).strip()}"
+            )
+
+    @abstractmethod
+    def _get_rates(self) -> Optional[ExchangeRates]:
+        """Actual implementation of the exchange rate getter."""
+        raise NotImplementedError
+
+
+class ApiExchangeRate(ExchangeRateGetter):
```

Comment:
I believe we should ditch this way of retrieving the rates using an external API all together.

Reply:
Understandable. I saw the comment mentioning other apps are also favoring the user-defined rates - and I get the simplicty is the key and not worth maintaining part of the app which would be used just by small partition of its users (providing API keys) .

Will remove.